core/src/main/javascript/reportviewer/reportviewer-prompt.js

```diff
@@ -652,7 +652,11 @@ define([
           var counter = 0;
           for (var i in param.values) {
             var paramVal = param.values[i];
-            if (value.indexOf(paramVal.value) > -1) {
+            if (value.indexOf(paramVal.value) > -1
+              //Datepickers require some extra care (this is needed if the values list has a stored default value and
+              // keeps it ready and selected=false.)
+              || (param.attributes['parameter-render-type'] === 'datepicker'
+                && this._compareDatesOnly(paramVal.value, value[0], param.timezoneHint, this._createFormatter(param)))) {
```

Comment:
Shouldn't you be comparing each position of the `value` array with `paramVal.value`? You're only comparing the first position.
Maybe your comment explains this, but I couldn't understand it.

Reply:
Further comment added. In the case where that checks occurs, the value object is a selected date, but is wrapped as an array with 1 object. So we only need to check the first item to get the selected value. The paramVal.val comes from the array of currently selected/default values and that is looped properly.

Reply:
Oh, I see now the reason why you only need to check the first position - the data picker does not allow multiple selection.